internal/server/index.go

```diff
@@ -36,9 +36,9 @@ func ServeIndex(publicDir string) fasthttp.RequestHandler {
 		nonce := utils.RandomString(32, alphaNumericRunes)
 
 		ctx.SetContentType("text/html; charset=utf-8")
-		ctx.Response.Header.Add("Content-Security-Policy", fmt.Sprintf("default-src 'self'; style-src 'self' 'nonce-%s'", nonce))
+		ctx.Response.Header.Add("Content-Security-Policy", fmt.Sprintf("base-uri 'none'; default-src 'self'; object-src 'none'; require-trusted-types-for 'script'; script-src 'self' 'unsafe-inline' 'nonce-%s'; style-src 'self' 'nonce-%s'", nonce, nonce))
```

Comment:
`'unsafe-inline'` and `require-trusted-types-for 'script'` are regression to the policy. I suggest to either use the nonce in the script tag or set the base value in an html tag to avoid modifying the CSP.

Reply:
Apparently the nonce is provided to the script tag so I guess those 2 attributes are not needed anymore.

Reply:
>Specifying nonce makes a modern browser ignore 'unsafe-inline' which could still be set for older browsers without nonce support.

`'unsafe-inline'` is [ignored if a nonce is provided in CSP2+ supported browser](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src) the only reason it was added was for browser backwards compatibility.

Perhaps my understanding of `require-trusted-types-for 'script'` was incorrect, [here is](https://web.dev/trusted-types/) what I was basing that change off of. I don't seem to understand how that's a regression in the policy.

Reply:
My understanding was not good, I tried to read the w3c documentation and I missed some of the essence. Your document explains it better, thanks for sharing. So you're right, it's not a regression, it's the contrary, we lock down some functions. Do you think it's safe? Won't we break some users?